Upstream: yes

commit 4752e41ff6235afaef88211e25e240df40ecb712
Author: Sam Lade <sam@sentynel.com>
Date:   Sat May 5 13:26:56 2012 +0100

    MusicBrainz: Deal with functions removed from ffmpeg
    
    ffmpeg 0.11 removed av_open_input_file, and current git libav removed
    av_find_stream_info and avcodec_open. #if uses of these functions to
    build on new ffmpeg/libav without removing ffmpeg 0.6 compatibility.
    There's still lots of deprecation warnings, but it builds.
    
    Thanks to Christoph Feck for pointing out the release of ffmpeg 0.11 broke
    av_open_input_file and the toAscii() -> toLocal8Bit() fix.
    
    REVIEW:105073

diff --git a/src/musicbrainz/MusicDNSAudioDecoder.cpp b/src/musicbrainz/MusicDNSAudioDecoder.cpp
index 0fa1946..97af62d 100644
--- a/src/musicbrainz/MusicDNSAudioDecoder.cpp
+++ b/src/musicbrainz/MusicDNSAudioDecoder.cpp
@@ -135,14 +135,21 @@ MusicDNSAudioDecoder::run()
 
     foreach( Meta::TrackPtr track, m_tracks )
     {
-        //TODO replace with "avformat_open_input" since av_open_input_file is deprecated
-        if( av_open_input_file( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toAscii(), NULL, 0, NULL ) )
+#if LIBAVFORMAT_VERSION_MAJOR >= 53
+        if( avformat_open_input( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toLocal8Bit(), NULL, NULL ) )
+#else
+        if( av_open_input_file( &pFormatCtx, ( const char * )track->playableUrl().toLocalFile().toLocal8Bit(), NULL, 0, NULL ) )
+#endif
         {
             warning() << QLatin1String( "Unable to open input file: " ) + track->playableUrl().toLocalFile();
             continue;
         }
 
+#if LIBAVFORMAT_VERSION_MAJOR >= 54
+        if( avformat_find_stream_info( pFormatCtx, NULL ) < 0 )
+#else
         if( av_find_stream_info( pFormatCtx ) < 0 )
+#endif
         {
             warning() << QLatin1String( "Unable to find stream info: " ) + track->playableUrl().toLocalFile();
             av_close_input_file( pFormatCtx );
@@ -173,7 +180,11 @@ MusicDNSAudioDecoder::run()
             continue;
         }
 
+#if LIBAVCODEC_VERSION_MAJOR >= 54
+        if( avcodec_open2( pCodecCtx, pCodec, NULL ) < 0 )
+#else
         if( avcodec_open( pCodecCtx, pCodec ) < 0 )
+#endif
         {
             warning() << QLatin1String( "Unable to open codec " ) + track->playableUrl().toLocalFile();
             av_close_input_file( pFormatCtx );
@@ -259,4 +270,4 @@ MusicDNSAudioDecoder::run()
 }
 
 
-#include "MusicDNSAudioDecoder.moc"
\ No newline at end of file
+#include "MusicDNSAudioDecoder.moc"
